### PR TITLE
docs: Add pages for `Script`s

### LINKED ISSUE #644: Add Docs Page: Scripts
### Motivation

Showcase how to use scripts, see also Rust SDK docs where applicable for examples

### Usage example

1. Use scripts
2. Use coins/messages with scripts
3. Add code examples

Helper that calls a contract using a script:
https://github.com/FuelLabs/fuels-ts/blob/master/packages/contract/src/scripts.ts
https://github.com/FuelLabs/fuels-ts/blob/master/packages/script/src/script.ts#L71

Direct script usage:
https://github.com/FuelLabs/fuels-ts/blob/master/packages/script/src/script.test.ts#L129-L142

### Possible implementations

Can probably borrow some copy from here as well: https://fuellabs.github.io/sway/v0.31.3/sway-program-types/scripts.html

### PR DESCRIPTION
**Pull request by Dhaiwat10**
Closes #644

### CODE REVIEW

File: docs/guide/scripts/calling-a-script.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
+---
+title: "Calling a script"
+parent: "Scripts"
+grand_parent: "Guide"
+nav_order: 2
+---
+
+# Calling a script
```
[the remark this answers is not in the document]
Comment (Dhaiwat10) [reply docs/guide/scripts/calling-a-script.md:8]: @camsjams gotcha, will keep this in mind

**Comment by Dhaiwat10** [comment]
@FuelLabs/sdk-ts I have added some very basic documentation around scripts which should help people at least get started with using them via our SDK. 

- I am still a little unclear on some details (eg. the `encoder` and `decoder` params in the `Script` constructor) so I haven't explained them for now. Hopefully the comments in the code snippets should compensate for the time being.

- The code snippet that I have used to explain the instantiation of a script is a little problematic since the code is scattered around a `beforeAll` code block. Not sure what the best move is in this scenario since we are directly referencing unit testing code through typedoc. The snippet can come off as confusing.

**Review by arboleya — APPROVED** [review]
LGTM.

I agree that the snippet around the `beforeAll` hook is a bit weird, but I think you did well for now, which put us in the right direction.

> **[sidenote]** Although we need unit-tested examples, it's important to realize that they don't need to be **_the_** unit tests themselves. In the long run, we can build a fully functional and tested App from the documentation perspective and then gather snippets from its **source code** — _not the unit tests_.

**Comment by arboleya** [comment]
@Dhaiwat10 Conflicts to resolve.

**Review by arboleya — APPROVED** [review]

File: docs/_guide/scripts/index.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+[nav_order: 7]
+
+# Scripts
+
+A Script, in Sway, is runnable bytecode on the chain which executes once to perform some task. A script can return a single value of any type.
+
+Learn more about Scripts [here](https://fuellabs.github.io/sway/v0.31.3/sway-program-types/scripts.html#scripts-and-the-sdks).
```
[the remark this answers is not in the document]
Comment (arboleya) [reply docs/_guide/scripts/index.md:7]: Fantastic tip.

**Review by arboleya — APPROVED** [review]

**Pull request merged**